### ISSUE TITLE
Error in HAPropsSI when using enthalpy as an input (Excel VBA)

### ISSUE THREAD

**Issue by ArchieLuxton** [CoolProp/CoolProp#2025]
I'm having some problems with the Excel wrapper calculating values based on Tdb and H - when I interrogate the error in the VBA console I can see that the HAPropsSI_temp function is returning either **-1.#IND** error or **-1.#QNAN** error.

Example inputs that give these errors are attached as a file, but a few are listed below:

Inputs: Tdb = 302.55, H = 64,654.35, P = 101,800
Output: -1.#IND

Inputs: Tdb = 283.15, H = 20,000, P = 101,400
Ouput = -1.#IND

Inputs: Tdb = 283.15, H = 20,000, P = 101,400 
Output = -1.#QNAN 

I've tried the following versions of the CoolProp_stdcall.dll source file (picked at random) and get the same errors: 

6.4.1
6.4.0
6.1.0
4.2.5
3.0.0

The DLL file is stored in my Temp location (_C:\Users\ME\AppData\Local\Temp_), but I have no problems performing other calculations so I don't think this is the problem.

I'm using the latest version of Excel on Microsoft 365, 32-bit. Windows 10 on a work computer. 

Many thanks

[CoolProp_Errors.xlsx](https://github.com/CoolProp/CoolProp/files/6596761/CoolProp_Errors.xlsx)

![QNAN](https://user-images.githubusercontent.com/78753319/120768779-6b96e100-c514-11eb-864a-46156267ca95.jpg)

![IND](https://user-images.githubusercontent.com/78753319/120768799-6fc2fe80-c514-11eb-93ba-4c5947490302.jpg)

**Comment by ibell** [comment]
How about atmospheric pressure? Same issue?

**Comment by ArchieLuxton** [comment]
Hi Ian, thanks for the response. I've attached a file that cycles through different atmospheric pressures, from 100,000 Pa to 101,480. 
[](url)
At certain points (roughly every ~200 Pa apart) it resolves the right value, but at the majority of pressures (and particularly around 100,300 - 100,560) it springs up the same errors. 

[CoolProp_Errors_Pressure.xlsx](https://github.com/CoolProp/CoolProp/files/6668521/CoolProp_Errors_Pressure.xlsx)

Thanks

**Comment by ibell** [comment]
Have you tried with the development version?  I have gone through a number
of campaigns of improving the convergence of HAPropsSI.  A recent one was I
think after 6.4.1 was released.

>

**Comment by ArchieLuxton** [comment]
Hi Ian,

So I've got to the bottom of it - turns out the CoolProp script in VBA was looking in a different place than I thought for the CoolProp_stdcall.dll file! It was looking by default in C:\Users\***\AppData\Roaming\CoolProp, even though I was setting the ChDir as my Temp files location. I added an extra line in the HAPropsSI function to check the current directory and set to Temp if it wasn't already. Not ideal to do this for thousands of calculations but seems to be reliable in making sure CoolProp looks in the right place for the dll (this reference has to be dynamic as other users use the same sheets/plugins).

The nightly release now works perfectly, thanks for the help!
